### PR TITLE
Core/Auras: implement Aura 379 (SPELL_AURA_MOD_MANA_REGEN_PCT)

### PR DESCRIPTION
**Pull request by DoctorKraft**
**Changes proposed:**

- Update mana regen with amount based on GT
- Remove old system with calculation based on spirit
- Implent Aura 379 (SPELL_AURA_MOD_MANA_REGEN_PCT)
Found on each passive spell of class/spec

**Target branch(es):** 3.3.5/master

- [X] master

### CODE REVIEW

**Comment by Hadifaren** [comment]
Doctor Do you know how to implement aura AURA_MOD_MAX_POWER 418

**Comment by Naios** [comment]
Is there anything missing for this PR? Otherwise it could be merged I think.

**Comment by DoctorKraft** [comment]
@Naios All are good for me, works well

File: src/server/game/Entities/Unit/StatSystem.cpp [hunk 1]
```diff
@@ -743,19 +744,24 @@ void Player::UpdateManaRegen()
...
+
+    // Apply PCT bonus from SPELL_AURA_MOD_MANA_REGEN_PCT
+    base_regen *= GetTotalAuraMultiplierByMiscValue(SPELL_AURA_MOD_MANA_REGEN_PCT, POWER_MANA);
+
+    // Apply PCT bonus from SPELL_AURA_MOD_MANA_REGEN_INTERRUPT
     int32 modManaRegenInterrupt = GetTotalAuraModifier(SPELL_AURA_MOD_MANA_REGEN_INTERRUPT);
 
-    SetStatFloatValue(UNIT_FIELD_POWER_REGEN_INTERRUPTED_FLAT_MODIFIER + manaIndex, base_regen + CalculatePct(spirit_regen, modManaRegenInterrupt));
-    SetStatFloatValue(UNIT_FIELD_POWER_REGEN_FLAT_MODIFIER + manaIndex, 0.001f + spirit_regen + base_regen);
+    SetStatFloatValue(UNIT_FIELD_POWER_REGEN_INTERRUPTED_FLAT_MODIFIER + manaIndex, base_regen * modManaRegenInterrupt);
```
Comment (Shauren) [inline src/server/game/Entities/Unit/StatSystem.cpp]: This is definitely wrong. SPELL_AURA_MOD_MANA_REGEN_INTERRUPT is a percentage (but im sure this aura isnt used anymore for calculating this field, all chars i tested had equal values in both)

Comment (DoctorKraft) [reply src/server/game/Entities/Unit/StatSystem.cpp]: Yeah i agree, but i have found some spells with this aura for NPC. Maybe move this aura for unit (creature) only? or remove?

Comment (Shauren) [reply src/server/game/Entities/Unit/StatSystem.cpp]: Remove it

Comment (DoctorKraft) [reply src/server/game/Entities/Unit/StatSystem.cpp]: Done

File: src/server/game/Entities/Unit/StatSystem.cpp [hunk 2]
```diff
@@ -743,19 +744,21 @@ void Player::UpdateManaRegen()
...
+    // Apply PCT bonus from SPELL_AURA_MOD_POWER_REGEN_PERCENT
+    base_regen *= GetTotalAuraMultiplierByMiscValue(SPELL_AURA_MOD_POWER_REGEN_PERCENT, POWER_MANA);
 
-    SetStatFloatValue(UNIT_FIELD_POWER_REGEN_INTERRUPTED_FLAT_MODIFIER + manaIndex, base_regen + CalculatePct(spirit_regen, modManaRegenInterrupt));
-    SetStatFloatValue(UNIT_FIELD_POWER_REGEN_FLAT_MODIFIER + manaIndex, 0.001f + spirit_regen + base_regen);
+    // Apply PCT bonus from SPELL_AURA_MOD_MANA_REGEN_PCT
+    base_regen *= GetTotalAuraMultiplierByMiscValue(SPELL_AURA_MOD_MANA_REGEN_PCT, POWER_MANA);
+
+    SetStatFloatValue(UNIT_FIELD_POWER_REGEN_INTERRUPTED_FLAT_MODIFIER + manaIndex, base_regen);
+    SetStatFloatValue(UNIT_FIELD_POWER_REGEN_FLAT_MODIFIER + manaIndex, base_regen);
```
Comment (joschiwald) [inline src/server/game/Entities/Unit/StatSystem.cpp]: regen and interrupted regen is always the same?

File: src/server/game/Spells/Auras/SpellAuraEffects.cpp [hunk 3]
```diff
@@ -444,7 +444,7 @@ pAuraEffectHandler AuraEffectHandler[TOTAL_AURAS]=
     &AuraEffect::HandleNULL,                                      //376 SPELL_AURA_MOD_CURRENCY_GAIN_FROM_SOURCE
     &AuraEffect::HandleNULL,                                      //377 SPELL_AURA_CAST_WHILE_WALKING_2
     &AuraEffect::HandleNULL,                                      //378
-    &AuraEffect::HandleNULL,                                      //379
+    &AuraEffect::HandleNoImmediateEffect,                         //379 SPELL_AURA_MOD_MANA_REGEN_PCT implemented in Player::UpdateManaRegen
```
Comment (joschiwald) [inline src/server/game/Spells/Auras/SpellAuraEffects.cpp]: i think on apply/remove you should call UpdateManaRegen

File: src/server/game/Entities/Unit/StatSystem.cpp (hunk 2 above)
Comment (DoctorKraft) [reply src/server/game/Entities/Unit/StatSystem.cpp]: Yes, shauren has also mentionned this in top "all chars i tested had equal values in both"

File: src/server/game/Spells/Auras/SpellAuraEffects.cpp (hunk 3 above)
Comment (DoctorKraft) [reply src/server/game/Spells/Auras/SpellAuraEffects.cpp]: rly? because they fields are correctly updated after a change spec

Comment (joschiwald) [reply src/server/game/Spells/Auras/SpellAuraEffects.cpp]: what happen if such aura is applied spec change independent

File: src/server/game/Entities/Unit/StatSystem.cpp (hunk 2 above)
Comment (joschiwald) [reply src/server/game/Entities/Unit/StatSystem.cpp]: what happend to SPELL_AURA_MOD_MANA_REGEN_INTERRUPT, there are still auras of that type

Comment (Shauren) [reply src/server/game/Entities/Unit/StatSystem.cpp]: iirc there are only 4 and none of them are usable

File: src/server/game/Spells/Auras/SpellAuraEffects.cpp (hunk 3 above)
Comment (DoctorKraft) [reply src/server/game/Spells/Auras/SpellAuraEffects.cpp]: Fixed

**Pull request merged**